src/databricks/labs/ucx/runtime.py

```diff
@@ -581,6 +584,35 @@ def scan_tables_in_mounts_experimental(
     ).snapshot()
 
 
+@task("all")
+def parse_logs(cfg: WorkspaceConfig, ws: WorkspaceClient, sql_backend: SqlBackend, _install: Installation):
+    """Parse the logs after workflow completion."""
+    _ = ws
+
+    task_name_format = r"^(\w+)\.log"
+    task_name_pattern = re.compile(task_name_format)
+
+    logs_recorder = LogsRecorder(
+        _install.install_folder(),
+        # Workaround by setting attributes in `run_task` to retrieve workflow ids
+        cfg.dynamically_set_attribut__workflow,  # type: ignore
+        cfg.dynamically_set_attribut__job_id,  # type: ignore
+        cfg.dynamically_set_attribut__workflow_run_id,  # type: ignore
+        sql_backend,
+        cfg.inventory_database,
+    )
+
+    for log_file in logs_recorder.log_path.glob("*.log"):
+        match = task_name_pattern.match(log_file.stem)
+        assert match is not None
+        task_name = match.group()
+
+        log_creation_timestamp = datetime.datetime.utcfromtimestamp(log_file.stat().st_ctime)
+
+        with log_file.open("r") as f:
+            logs_recorder.record(task_name, f, log_creation_timestamp)
```

Comment:
make this a method in TaskRunWarningsRecorder. tasks have to be extremely thin and just call methods of objects in the runtime context.